internal/utils/utils.go

```diff
@@ -21,6 +29,23 @@ func CominServiceRestart() error {
 	return nil
 }
 
+func cominServiceRestartDarwin() error {
+	logrus.Infof("Comin service configuration changed - will restart after deployment")
+	
+	// On Darwin, create a flag file that signals the main process should exit
+	// after the current deployment completes. Launchd will automatically restart
+	// the process due to KeepAlive=true, which is more reliable than self-restart
+	
+	restartFlagPath := "/var/lib/comin/restart-required"
```

Comment:
I don't understand why do you need to create this file. Why is it not enough to just exit comin when a restart is needed?